module/item/entity.js

```diff
@@ -1907,8 +1907,8 @@ export default class Item5e extends Item {
 
     // NPCs automatically equip items and are proficient with them
     if ( isNPC ) {
-      updates["data.equipped"] = true;
-      updates["data.proficient"] = true;
+      if ( foundry.utils.getProperty(data, "data.equipped") === undefined ) updates["data.equipped"] = true;
+      if ( foundry.utils.getProperty(data, "data.proficient") === undefined ) updates["data.proficient"] = true;
```

Comment:
For v10 these will need to point to `system`, plus you can probably make it shorter using optional chaining rather than getProperty:

```suggestion
      if ( data.system?.equipped === undefined ) updates["system.equipped"] = true;
      if ( data.system?.proficient === undefined ) updates["system.proficient"] = true;
```

Reply:
The `getProperty` was to ensure that these two cases are equivalent:

```js
(async () => {
	const item1 = await Item.create({
		name: "Item 1",
		type: "weapon",
		// property path as string
		"system.proficient": true,
	});
	console.log(item1.system.proficient);

	const item2 = await Item.create({
		name: "Item 2",
		type: "weapon",
		// property path as nested objects
		system: {proficient: true},
	});
	console.log(item2.system.proficient);

	// ////////////////

	item1.sheet.render(true);
	item2.sheet.render(true);
})();
```

Is there some magic I'm missing here that renders that unnecessary?

Reply:
You are correct, forgot that non-expanded data can appear in this updates :).